modules/auxiliary/scanner/smb/smb_uninit_cred.rb

```diff
@@ -0,0 +1,275 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'msf/core/auxiliary/report'
+
+class Metasploit3 < Msf::Auxiliary
+
+  # Exploit mixins should be called first
+  include Msf::Exploit::Remote::DCERPC
+  include Msf::Exploit::Remote::SMB::Client
+  include Msf::Exploit::Remote::SMB::Client::Authenticated
+
+  # Scanner mixin should be near last
+  include Msf::Auxiliary::Scanner
+  include Msf::Auxiliary::Report
+
+  # Aliases for common classes
+  SIMPLE = Rex::Proto::SMB::SimpleClient
+  XCEPT  = Rex::Proto::SMB::Exceptions
+  CONST  = Rex::Proto::SMB::Constants
+
+  RPC_NETLOGON_UUID = '12345678-1234-abcd-ef00-01234567cffb'
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => 'Samba _netr_ServerPasswordSet Uninitialized Credential State',
+      'Description'    => %q{
+        This module checks if your Samba target is vulnerable to an uninitialized variable creds.
+      },
+      'Author'         =>
+        [
+          'Richard van Eeden', # Original discovery
+          'sleepya',           # Public PoC for the explicit check
+          'sinn3r'
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          ['CVE', '2015-0240'],
+          ['OSVDB', '118637'],
+          ['URL', 'https://securityblog.redhat.com/2015/02/23/samba-vulnerability-cve-2015-0240/'],
+          ['URL', 'https://gist.github.com/worawit/33cc5534cb555a0b710b'],
+          ['URL', 'https://www.nccgroup.com/en/blog/2015/03/samba-_netr_serverpasswordset-expoitability-analysis/']
+        ],
+      'DefaultOptions' =>
+        {
+          'SMBDirect'               => true,
+          'SMBPass'                 => '',
+          'SMBUser'                 => '',
+          'SMBDomain'               => '',
+          'DCERPC::fake_bind_multi' => false
+        }
+    ))
+
+    # This is a good example of passive vs explicit check
+    register_options([
+      OptEnum.new('CHECK_STYLE',
+        [
+          true,
+          'Explicit style will actually try to trigger the bug, otheriwse purely a banner check',
+          'PASSIVE',
+          ['EXPLICIT', 'PASSIVE']
```

Comment:
I can make it OptBool, and I'll make "Explicit" as default.

Accuracy is more of a better choice to ensure quality.